pkg/api/unversioned/group_version.go

```diff
@@ -241,6 +253,28 @@ func (gv *GroupVersion) UnmarshalText(value []byte) error {
 	return gv.unmarshal(value)
 }
 
+// GroupVersions can be used to represent a set of desired group versions.
+type GroupVersions []GroupVersion
+
+// VersionForGroup implements the same interface as GroupVersions to allow easy
+// selection of group versions.
+func (gvs GroupVersions) VersionForGroup(group string) (GroupVersion, bool) {
```

Comment:
I'll add an assertion to runtime - that's just as effective.

On Mon, May 23, 2016 at 11:23 AM, David Eads <notifications@github.com>
wrote:

> In pkg/api/unversioned/group_version.go
> <https://github.com/kubernetes/kubernetes/pull/26044#discussion_r64240099>
> :
>
> > @@ -241,6 +253,28 @@ func (gv *GroupVersion) UnmarshalText(value []byte) error {
> >  	return gv.unmarshal(value)
> >  }
> >
> > +// GroupVersions can be used to represent a set of desired group versions.
> > +type GroupVersions []GroupVersion
> > +
> > +// VersionForGroup implements the same interface as GroupVersions to allow easy
> > +// selection of group versions.
> > +func (gvs GroupVersions) VersionForGroup(group string) (GroupVersion, bool) {
>
> Interfaces can be defined in multiple spots. Typing by coincidence is
> really annoying. On Mon, May 23, 2016 at 11:20 AM, Clayton Coleman <
> notifications@github.com> wrote:
> … <#m_1495268411925230081_>
> In pkg/api/unversioned/group_version.go <#26044 (comment)
> <https://github.com/kubernetes/kubernetes/pull/26044#discussion_r64239638>>
> : > @@ -241,6 +253,28 @@ func (gv *GroupVersion) UnmarshalText(value
> []byte) error { > return gv.unmarshal(value) > } > > +// GroupVersions can
> be used to represent a set of desired group versions. > +type GroupVersions
> []GroupVersion > + > +// VersionForGroup implements the same interface as
> GroupVersions to allow easy > +// selection of group versions. > +func (gvs
> GroupVersions) VersionForGroup(group string) (GroupVersion, bool) { I don'
> … <#m_4800330639593875586_> On Mon, May 23, 2016 at 11:17 AM, David Eads
> ***@***.***> wrote: In pkg/api/unversioned/group_version.go <#26044
> <https://github.com/kubernetes/kubernetes/pull/26044> (comment) <#26044
> (comment)
> <https://github.com/kubernetes/kubernetes/pull/26044#discussion_r64239278>>>
> : > @@ -241,6 +253,28 @@ func (gv *GroupVersion) UnmarshalText(value
> []byte) error { > return gv.unmarshal(value) > } > > +// GroupVersions can
> be used to represent a set of desired group versions. > +type GroupVersions
> []GroupVersion > + > +// VersionForGroup implements the same interface as
> GroupVersions to allow easy > +// selection of group versions. > +func (gvs
> GroupVersions) VersionForGroup(group string) (GroupVersion, bool) { declare
> an actual interface here for VersionForGroup and PrefersGroup. Otherwise I
> think its less than obvious. — You are receiving this because you authored
> the thread. Reply to this email directly or view it on GitHub <
> https://github.com/kubernetes/kubernetes/pull/26044/files/443d00fbad3785e2c68386cd4de38344f1ee21f6..9d83ae05067c2ad7bb5d3eda95d3fc3aadd77123#r64239278>
> — You are receiving this because you were assigned. Reply to this email
> directly or view it on GitHub <
> https://github.com/kubernetes/kubernetes/pull/26044/files/443d00fbad3785e2c68386cd4de38344f1ee21f6..9d83ae05067c2ad7bb5d3eda95d3fc3aadd77123#r64239638>
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26044/files/443d00fbad3785e2c68386cd4de38344f1ee21f6..9d83ae05067c2ad7bb5d3eda95d3fc3aadd77123#r64240099>
>